### DISCUSSION TITLE
Option to Disable Wi-Fi?

### DISCUSSION THREAD

**Discussion by AlaskanAstro** [aaronwmorris/indi-allsky#1406] [Ideas]
Hello, just noticed that the culprit for randomly poor video loading performance was (I'm pretty sure), the pi connecting over my wifi network randomly instead of its hardwired ethernet. I used rfkill to disable wifi and now it's as fast as ever when I connect via the 'devicename'.local address. 

Perhaps it would be useful to others to have that option baked into indi-allsky? Or maybe even just a mention in the install guide? Or maybe that wasn't my issue and it'll be slow again tomorrow, haha.

**Comment by aaronwmorris** [comment]
This would only be a problem is you ever connect via wifi and have ethernet connected.  Some people run and view their cameras entirely via WIFI.  WIFI is one of those things that, it works until it doesn't :-)

I try to recommend people to use ethernet (and PoE) where possible, however, for the most part I just try to "meet people where they are".  One of the traps I try to avoid is telling people how they should use their systems.  Bad network performance due to wifi just does not bother some people.

**Reply by AlaskanAstro** [reply]
Oh for sure, I'm not saying that Ethernet only should be pushed as any kind of "primary" way. More that it would be nice to have the option to disable wifi for those of us (maybe just me), who set up in the house over wifi then deploy permanently on Ethernet and want to disable it to avoid the random wifi connection. 

Also to be fair, it took me maybe 30 seconds of Googling to find a terminal command to do it myself so maybe this is unnecessary.

**Comment by Vertikar** [comment]
I've got this script running on a few of my Raspberry Pi's where I want fallback to Wifi to be automatic.
Ethernet is preferred and if its available it'll disable wifi automattically.

https://blog.christophersmart.com/2021/11/02/automatically-enable-and-disable-wifi-based-on-ethernet-connection-with-networkmanager/
